app/notification.go

```diff
@@ -15,13 +15,138 @@ import (
 	"github.com/mattermost/mattermost-server/store"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/mattermost/mattermost-server/utils/markdown"
+	"github.com/mitchellh/mapstructure"
 )
 
 const (
 	THREAD_ANY  = "any"
 	THREAD_ROOT = "root"
 )
 
+// TODO : Comment
+func (a *App) SendEphemeralPostWrapper(userId string, postDetails interface{}, post model.Post) {
```

Comment:
I don't understand why this map structure is needed, I don't see any reason to not use directly the `model.Post` struct.

Reply:
The mapstructure is tightly coupled with the comment you left below. It takes an interface and turns it into Post type. So if i change the code in the comment below, the mapstructure along with the `SendEphemeralPostWrapper` will be removed.